llvm/lib/Transforms/Instrumentation/MemProfiler.cpp

```diff
@@ -795,8 +795,8 @@ struct AllocMatchInfo {
   bool Matched = false;
 };
 
-DenseMap<uint64_t, SmallVector<CallEdgeTy, 0>>
-memprof::extractCallsFromIR(Module &M) {
+DenseMap<uint64_t, SmallVector<CallEdgeTy, 0>> memprof::extractCallsFromIR(
+    Module &M, function_ref<const TargetLibraryInfo &(Function &)> GetTLI) {
```

Comment:
The only reason I am passing `GetTLI` is because `getResult` takes a function:

```
FAM.getResult<TargetLibraryAnalysis>(F);
```

I don't know if `TLI` actually depends on the function being passed.

Reply:
I think you can use the WrapperPass based approach on L117 here: https://github.com/llvm/llvm-project/commit/2e426fe8ff314c2565073e73e27fdbdf36c140a3#diff-b13f32dcb9d84dd5b7c4d5964bad73df362fbbe51ac3e551765aca1e9299c2b6R117

Reply:
Thanks!  I've incorporated `TargetLibraryInfoWrapperPass` in the latest revision.